src/ripple/shamap/SHAMap.h

```diff
@@ -82,7 +82,7 @@ class SHAMap
     Family&                         f_;
     beast::Journal                  journal_;
     std::uint32_t                   seq_;
-    std::uint32_t                   ledgerSeq_; // sequence number of ledger this is part of
+    std::uint32_t mutable           ledgerSeq_; // sequence number of ledger this is part of
```

Comment:
@nbougalis It works because `fetchNodeFromDB` is doing a `const_cast` (https://github.com/nbougalis/rippled/blob/ledger/src/ripple/shamap/impl/SHAMap.cpp#L176). We need either the `mutable` or the `const_cast`, but not both.

Things like `Ledger::setFull` because they manipulate the `SHAMap` through a `shared_ptr` (https://github.com/seelabs/rippled/blob/8389f95fa9e0cadeaf01c9663d61e8e32ffa08dc/src/ripple/app/ledger/Ledger.h#L259-L26)

I still prefer removing the `mutable` and making `setLedgetSeq` non-const. The `const_cast` makes clear the one place where `ledgerSeq_` is a problem.